### PR TITLE
Use user obj / pull channel name/avatar from server

### PR DESCRIPTION
**Pull request by neuroscr**
- Addresses #470 (using the user object server object to pull pubkey and nickname instead of individual message annotation)
- Also set up pulling channel meta data (name, avatar) from server (channel annotation) on a regular basis
- other minor refactors

### CODE REVIEW

File: js/modules/loki_public_chat_api.js [hunk 1]
```diff
@@ -312,8 +347,11 @@ class LokiPublicChannelAPI {
       log.info(`deleted ${serverId} on ${this.baseChannelUrl}`);
       return true;
     }
+    // fire an alert
     log.warn(`failed to delete ${serverId} on ${this.baseChannelUrl}`);
-    return false;
+    throw new textsecure.PublicChatError(
```
Comment (BeaudanBrown) [inline js/modules/loki_public_chat_api.js]: Where we are calling this function in `js/views/conversation_view.js` line ~1300 we aren't expecting this to throw
So we either need to catch it there and stop returning a bool, or revert this change and keep the bool as the success/fail flag

File: js/modules/loki_public_chat_api.js [hunk 2]
```diff
@@ -326,41 +364,36 @@ class LokiPublicChannelAPI {
...
-        response,
-      };
+    const res = await this.serverRequest(`${this.baseChannelUrl}`, {
+      params: {
+        include_annotations: 1,
+      },
+    });
+    if (!res.err && res.response) {
+      if (
+        res.response.data.annotations &&
```
Comment (BeaudanBrown) [inline js/modules/loki_public_chat_api.js]: Can we just collapse these 2 if's to 
```js
if (
  !res.error &&
  res.response &&
  res.response.data &&
  res.response.data.annotations &&
  res.response.data.annotations.length !== 0
) {
...
```

File: js/modules/loki_public_chat_api.js [hunk 3]
```diff
@@ -203,18 +204,21 @@ class LokiPublicChannelAPI {
     this.serverAPI = serverAPI;
     this.channelId = channelId;
     this.baseChannelUrl = `channels/${this.channelId}`;
-    this.groupName = 'unknown';
     this.conversationId = conversationId;
+    this.conversation = ConversationController.get(conversationId);
```
Comment (BeaudanBrown) [inline js/modules/loki_public_chat_api.js]: Will leave this for now and implement the fix discussed yesterday later

File: js/modules/loki_public_chat_api.js (hunk 1 above)
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: will remove for now and re-address as parameter later

File: js/modules/loki_public_chat_api.js (hunk 2 above)
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: resolved

File: js/modules/loki_public_chat_api.js (hunk 3 above)
Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: which was? a check or revert? or just making sure we stop timers and deallocate channel API too?

Comment (BeaudanBrown) [reply js/modules/loki_public_chat_api.js]: Ah just the discussion with Sach about grabbing this when we need it rather than storing a reference on the api object

Comment (neuroscr) [reply js/modules/loki_public_chat_api.js]: ok so a revert
